framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3785,23 +3785,25 @@ protected void checkTypeArguments(
             }
 
             checkHasQualifierParameterAsTypeArgument(typeArg, paramUpperBound, toptree);
+            Object paramName = paramNames.get(i);
+
             commonAssignmentCheck(
                     paramUpperBound,
                     typeArg,
                     reportErrorToTree,
                     "type.argument.type.incompatible",
-                    paramNames.get(i),
+                    paramName,
                     typeOrMethodName);
 
             if (!typeHierarchy.isSubtype(bounds.getLowerBound(), typeArg)) {
                 FoundRequired fr = FoundRequired.of(typeArg, bounds);
                 checker.reportError(
                         reportErrorToTree,
                         "type.argument.type.incompatible",
-                        paramNames.get(i),
+                        paramName,
                         typeOrMethodName,
                         fr.found,
-                        fr.required);
+                        paramName + fr.required);
```

Comment:
Should we add a space here, between the `paramName` and the bounds?

Are the square brackets useful? How do they interact with the flag to show more details about bounds?

Reply:
I like the output that is given here: https://github.com/eisop/checker-framework/blob/ec22f85d6490bfb3a711af9157bbf0f2747c5143/framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java#L342
That is, a space after the type parameter name and spaces in the square brackets.